### PR TITLE
Clarify allowable backend switches in matplotlib.use().

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

Per discussion at https://gitter.im/matplotlib/matplotlib?at=60e1fec28c12474d8ceab2a6.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/matplotlib/__init__.py [hunk 1]
```diff
@@ -1074,11 +1074,16 @@ def use(backend, *, force=True):
 
         or a string of the form: ``module://my.module.name``.
 
+        Switching to an interactive backend is impossible if an unrelated
```
Comment (timhoffm) [inline lib/matplotlib/__init__.py:1077]: Would slightly prefer: "impossible" -> "not supported" (or maybe "not possible"). But could also be merged as is.

Comment (anntzer) [reply lib/matplotlib/__init__.py]: Switched to "not possible".  It's really close to impossible because the GUI toolkits globally set the inputhook...

**Review by timhoffm — APPROVED** [review]
Anybody can merge after CI pass.

File: lib/matplotlib/__init__.py (hunk 1 above)
Comment (timhoffm) [reply lib/matplotlib/__init__.py]: Agreed. (But you don't want to discuss that with the end-user.)

**Pull request merged**